### PR TITLE
fix(Core/SmartScripts): Fixed `SMART_ACTION_START_FOLLOW` not complet…

### LINKED ISSUE #12153: [quest]`Find the Ancient Hero` bug
### Current Behaviour

The quest requires that `Subjugated Iskalder` be brought to `The Bone Witch`. After that, the quest cannot be completed

### Expected Blizzlike Behaviour

The quest requires that `Subjugated Iskalder` be brought to `The Bone Witch`. After that, the quest can be completed

### Source

_No response_

### Steps to reproduce the problem

1. .q a 13133
2. .go c id 30718
3. talk it
4. when he becomes `iskalder`, then use `The Bone Witch's Amulet` for `iskalder`
5. bring 'iskalder' back to 'The Bone Witch'

### Extra Notes

_No response_

### AC rev. hash/commit

38889324b2bd4a77b47bf4774753198e280723ec

### Operating system

Ubuntu20.04

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by UltraNix**
…ing quests.

Fixed completing "Find the Ancient Hero" quest.
Fixes #12153

## Issues Addressed:

- Closes #12153

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

`.go xyz 7658.467 3439.82 655.936 571`
`.quest add 13133`
`.npc add temp 30884`
Use amulet on mob
Go with Iskalder to Bone Witch and see if quest is completed.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -917,7 +917,7 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
                 if (IsUnit(target))
                 {
                     float angle = e.action.follow.angle > 6 ? (e.action.follow.angle * M_PI / 180.0f) : e.action.follow.angle;
-                    CAST_AI(SmartAI, me->AI())->SetFollow(target->ToUnit(), float(e.action.follow.dist) + 0.1f, angle, e.action.follow.credit, e.action.follow.entry, e.action.follow.creditType);
+                    CAST_AI(SmartAI, me->AI())->SetFollow(target->ToUnit(), float(e.action.follow.dist) + 0.1f, angle, e.action.follow.credit, e.action.follow.entry, e.action.follow.creditType, e.action.follow.aliveState);
```
Comment (pklloveyou) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:920]: ```e.action.follow.aliveState != 0```
better？

Comment (UltraNix) [reply src/server/game/AI/SmartScripts/SmartScript.cpp:920]: No.

**Comment by pklloveyou** [comment]
Tested

**Comment by mpfans** [comment]
After completing the quest, wait for `Subjugated Iskalder` to disappear and talk with `The Bone Witch`. The quert cannot be returned

**Comment by pklloveyou** [comment]
@mpfans 
image
You have to run back with it

**Comment by UltraNix** [comment]
@mpfans Just tested and works okay everything.

**Comment by pklloveyou** [comment]
You have to run back with it @mpfans

**Comment by mpfans** [comment]
This is a test video:
https://youtu.be/RixaEcLc7dQ

**Comment by UltraNix** [comment]
@mpfans Have you rewarded that quest before? Remove it from your questlg and then use  `.quest remove 13133`. And try again.

**Comment by mpfans** [comment]
ok ... `DELETE FROM character_queststatus_rewarded WHERE guid =1 AND quest = 13133;`

retested and work now.... tyvm

**Review by IntelligentQuantum — APPROVED** [review]

**Pull request merged**